### PR TITLE
rustdoc: tweak notes on ignore/compile_fail examples

### PR DESCRIPTION
**Pull request by QuietMisdreavus**
Part of https://github.com/rust-lang/rust/issues/44927

This is a softening of these notices to mention *why* a given example has a given callout, rather then telling viewers to be careful with an example. It also changes the character used for these samples from a warning logo to a circle-i/information logo.

image

### CODE REVIEW

**Comment by QuietMisdreavus** [comment]
@rust-lang/docs These strings are being slurped up again, what do you think of these updates?

File: src/librustdoc/html/markdown.rs [hunk 1]
```diff
@@ -228,9 +228,9 @@ impl<'a, I: Iterator<Item = Event<'a>>> Iterator for CodeBlocks<'a, I> {
                 ))
             });
             let tooltip = if ignore {
-                Some(("Be careful when using this code, it's not being tested!", "ignore"))
+                Some(("This example is not tested", "ignore"))
```
Comment (GuillaumeGomez) [inline src/librustdoc/html/markdown.rs]: I'm not a big fan of this new sentence. :-/

File: src/librustdoc/html/markdown.rs [hunk 2]
```diff
@@ -228,9 +228,9 @@ impl<'a, I: Iterator<Item = Event<'a>>> Iterator for CodeBlocks<'a, I> {
...
             } else if compile_fail {
-                Some(("This code doesn't compile so be extra careful!", "compile_fail"))
+                Some(("This example deliberately fails to compile", "compile_fail"))
```
Comment (GuillaumeGomez) [inline src/librustdoc/html/markdown.rs]: Nice!

File: src/librustdoc/html/markdown.rs (hunk 1 above)
Comment (bluss) [reply src/librustdoc/html/markdown.rs]: It's matter of fact, much better.

**Comment by QuietMisdreavus** [comment]
Oh right, there's a test for the warning text, so travis failed because i didn't change it:

```
[00:53:03] 15: @has check failed
[00:53:03] 	`XPATH PATTERN` did not match
[00:53:03] 	// @has foo/fn.bar.html '//*[@class="tooltip compile_fail"]/span' "This code doesn't compile so be extra careful!"
[00:53:03] 16: @has check failed
[00:53:03] 	`XPATH PATTERN` did not match
[00:53:03] 	// @has foo/fn.bar.html '//*[@class="tooltip ignore"]/span' "Be careful when using this code, it's not being tested!"
[00:53:03] 
[00:53:03] Encountered 2 errors
[00:53:03] 
[00:53:03] ------------------------------------------
[00:53:03] 
[00:53:03] thread '[rustdoc] rustdoc/codeblock-title.rs' panicked at 'explicit panic', /checkout/src/tools/compiletest/src/runtest.rs:2498:8
```

**Review by frewsxcv — APPROVED** [review]
lgtm! the text associated with the examples is a bit subjective, but i personally think this is an improvement. r=me with test fixes and if imperio is cool with the changes

**Comment by carols10cents** [comment]
triage ping for you @QuietMisdreavus! ❤️

**Comment by QuietMisdreavus** [comment]
I've pushed a commit that fixes the test.

@GuillaumeGomez Do you have any specific concerns about the new text, or are you okay with this landing? AFAIK @frewsxcv wanted to make sure you were okay with the changes before approving.

**Comment by GuillaumeGomez** [comment]
Oops sorry! The current text is fine for me so let's merge.

@bors: r+ rollup

**Pull request merged**